code/modules/mob/mob_movement.dm

```diff
@@ -157,6 +157,8 @@
 						M.stop_pulling()
 
 		switch(mob.m_intent)
+			if("sprint")
+				move_delay += config.run_speed

```

Comment:
Actually it would be better to have the damage affect your stamina instead of speed.